flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSpeculativeSinkITCase.java

```diff
@@ -0,0 +1,179 @@
+                            .name("slowMap")
+                            .returns(Types.ROW_NAMED(new String[] {"i", "j"}, Types.INT, Types.INT))
+                            .setParallelism(3);
+
+            Table t =
+                    tEnv.fromChangelogStream(
+                            ds,
+                            Schema.newBuilder()
+                                    .column("i", DataTypes.INT())
+                                    .column("j", DataTypes.INT())
+                                    .build(),
+                            ChangelogMode.insertOnly());
+
+            tEnv.getConfig().setSqlDialect(SqlDialect.DEFAULT);
+            tEnv.createTemporaryView("mappedTable", t);
+            String insertQuery = "insert into append_table select * from mappedTable";
+
+            // assert that the SlowMap operator is chained with the hive table sink, which will lead
+            // to a slow sink as well.
+            StreamTableEnvironmentImpl tEnvImpl = (StreamTableEnvironmentImpl) tEnv;
+            JobGraph jobGraph =
+                    tEnvImpl.execEnv()
+                            .generateStreamGraph(
+                                    tEnvImpl.getPlanner()
+                                            .translate(
+                                                    Collections.singletonList(
+                                                            (ModifyOperation)
+                                                                    tEnvImpl.getParser()
+                                                                            .parse(insertQuery)
+                                                                            .get(0))))
+                            .getJobGraph();
+
+            for (JobVertex jobVertex : jobGraph.getVertices()) {
+                if (jobVertex.getName().contains("slowMap")) {
+                    assertThat(jobVertex.getName().contains("Sink")).isTrue();
+                }
+            }
+            tEnv.executeSql(insertQuery).await();
+
+            List<Row> rows =
+                    CollectionUtil.iteratorToList(
+                            tEnv.executeSql("select * from append_table").collect());
+            rows.sort(Comparator.comparingInt(o -> (int) o.getField(0)));
+
+            // Finally we will get the output value with attemptNumber larger than 0
+            assertThat(rows).isEqualTo(Collections.singletonList(Row.of(0, 1)));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            tEnv.executeSql("drop database db1 cascade");
+        }
+    }
+
+    private Configuration configure(Configuration configuration) {
+        // for speculative execution
+        configuration.set(BatchExecutionOptions.SPECULATIVE_ENABLED, true);
+        // for testing, does not block node by default
+        if (!configuration.contains(BatchExecutionOptions.BLOCK_SLOW_NODE_DURATION)) {
```

Comment:
This `if` is not needed because `configure(...)` is a private method and is used only by this test.